DoublyLinkedList/doubly_linked_list.h

```diff
@@ -0,0 +1,288 @@
+    void clear();
+};
+//for  finding node at a position
+node *doubly_linked_list::find_node(int location)
+{
+    node *temp = head;
+    int count = 1;
+    //iterating to insert at a position
+    while (count != location)
+    {
+        // to check whether the location is present or not
+        //traversal till the condition is satisfied
+        temp = temp->next;
+        count++;
+    }
+    return (temp);
+}
+
+//adding element at the front
+void doubly_linked_list::push_front(int value)
+{
+    insert(value, 0);
+}
+//add the element at the back
+void doubly_linked_list::push_back(int value)
+{
+    insert(value, length - 1);
+}
+//for inserting at a location
+void doubly_linked_list::insert(int value, int location)
+{
+    if (location > length || location < 0)
+    {
+        cout << "Invalid Location specified to Insert" << endl;
+        return;
+    }
+
+    if (location == 0)
+    {
+        //new node creation
+        node *new_node = new node(nullptr, value, head);
+
+        //if head node is not empty then prev of head node will be assigned to the new node
+        if (head != nullptr)
+        {
+            head->prev = new_node;
+        }
+        //if the node is empty then both head and tail will point to the new node
+        else
+        {
+            tail = new_node;
+        }
+        //now the new node becomes the first node
+        head = new_node;
+        length++;
+        return;
+    }
+    if (location == length)
+    {
+        //new node creation
+        node *new_node = new node(tail, value, nullptr);
+
+        //if the list is not empty then tail node is not empty then next of tail node will be assigned to the new node
+        if (tail != nullptr)
+        {
+            tail->next = new_node;
+        }
+        else
+        {
+            head = new_node;
+        }
+        //now the new node becomes the last node
+        tail = new_node;
+        length++;
+        return;
+    }
+
+    //created temporary nodes which will point to previous and next node
+    node *temp1 = find_node(location);
+    //new node has to be inserted after the temp1 node then temp1 is the previous node
+    //temp2 node will be next node
+    node *temp2 = temp1->next;
+    //as the new node is inserted between two temp nodes,
+    node *new_node = new node(temp1, value, temp2);
+    //assigning values for the two before and after node.
+    temp1->next = new_node;
+    temp2->prev = new_node;
+    length++;
+    return;
+}
+
+//for removing a node at front in the list
+int doubly_linked_list::remove_front()
+{
+    return remove(0);
+}
+//for removing last item in the list
+int doubly_linked_list::remove_back()
+{
+    return remove(length - 1);
+}
+//for removing node at a position
+int doubly_linked_list::remove(int location)
+{
+    if (location >= length || location < 0)
+    {
+        cout << "Invalid Location Specified to Remove" << endl;
+        return 0;
+    }
+    if (location == 0)
```

Comment:
this function can be made better. for one thing we just need to delete the node, get its prev and next pointer. 

the special cases are 
1) when prev is null, then we are removing head,
2) when next is null, we are removing tail
(3) when both are null we are removing only element in list)

given this simplify the function.